eng/common/pipelines/templates/steps/docs-metadata-release.yml

```diff
@@ -0,0 +1,58 @@
+# intended to be used as part of a release process
+parameters:
+  ArtifactLocation: 'not-specified'
+  PackageRepository: 'not-specified'
+  ReleaseSha: 'not-specified'
+  RepoId: 'not-specified'
+  WorkingDirectory: ''
+  ScriptDirectory: eng/common/scripts
+  TargetDocRepo: ''
+  TargetDocBranch: 'smoke-test'
+  PRBranchName: 'not-specified'
+  ArtifactName: ''
+  ServiceDirectory: ''
+
+steps:
+- pwsh: |
+    git clone ${{ TargetDocRepository }} ${{ parameters.WorkingDirectory }}/repo
+    
+    $sourceBranch = "${{ parameters.TargetDocBranch }}"
+  
+    try {
+      Push-Location {{ parameters.WorkingDirectory }}/repo
+    
+      # 0 if exists, 1 if not
+      git show-ref --verify --quiet refs/heads/$sourceBranch
```

Comment:
Given you have a fresh clone a few steps above there should not be any existing branch other then master in the local clone. There might be a remote branch with the same name but what you are doing here isn't going to do what I think you want it to.

Reply:
Also if there does end up being a remote branch with the same name the forced reset at https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/git-branch-push.ps1#L108 will fix it up to rebase put the changes on top of that branch.

Do you expect that the branch will already exist in the normal case? If not then we should be good here but if so then we can make some changes to the git-branch-push script to look for a remote branch of the same name and base the checkout off of that if it exists.

Reply:
Ok. In a normal case we expect `smoke-test` to exist. always. Hard error if not.